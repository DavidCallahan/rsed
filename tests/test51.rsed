TABLE="machine_info"
where = {}
foreach past required "^power management"
   skip if "^processor"
   required "^(.*\S)\s*:(.*)" 
   name=replace " " with "_" in $1
   value=quote(trim($2))
   where = append($where, " $name = $value")
end
id = $TABLE "_id"
print "SELECT $id from $TABLE WHERE" join ("\n\tAND", $where)
