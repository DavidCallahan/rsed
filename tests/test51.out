SELECT machine_info_id from machine_info WHERE vendor_id = "GenuineIntel"
	AND cpu_family = "6"
	AND model = "45"
	AND model_name = "Intel(R) Xeon(R) CPU E5-2660 0 @ 2.20GHz"
	AND stepping = "6"
	AND microcode = "0x615"
	AND cpu_MHz = "2200.000"
	AND cache_size = "20480 KB"
	AND physical_id = "0"
	AND siblings = "16"
	AND core_id = "0"
	AND cpu_cores = "8"
	AND apicid = "0"
	AND initial_apicid = "0"
	AND fpu = "yes"
	AND fpu_exception = "yes"
	AND cpuid_level = "13"
	AND wp = "yes"
	AND flags = "fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc aperfmperf eagerfpu pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic popcnt tsc_deadline_timer aes xsave avx lahf_lm arat epb pln pts dtherm tpr_shadow vnmi flexpriority ept vpid xsaveopt"
	AND bugs = ""
	AND bogomips = "4400.37"
	AND clflush_size = "64"
	AND cache_alignment = "64"
	AND address_sizes = "46 bits physical, 48 bits virtual"
	AND power_management = ""
